tests/test_config_parsing/test_parser_error_collection.py

```diff
@@ -160,25 +161,26 @@ def assert_that_config_leads_to_warning(
     with pytest.warns(ConfigWarning) as the_warnings:
         ErtConfig.from_file(config_filename, use_new_parser=True)
 
-    warning_infos: List[WarningInfo] = []
+    warning_infos: List[ErrorInfo] = []
     for x in the_warnings.list:
-        assert isinstance(x, Warning)
-        warning_info: WarningInfo = x.args[0]
+        assert isinstance(x, WarningMessage)
+        message = cast(ConfigWarning, x.message)
```

Comment:
Is this to silence mypy? I fail to see what it does here, as you could have done `x.message.info` instead?